pkg/agents/image/image.go

```diff
@@ -79,25 +88,43 @@ func (a *agent) Start(ctx context.Context) {
 
 					select {
 					case <-ctx.Done():
+						// Ensure the timer channel is drained
+						if !timer.Stop() {
+							select {
+							case <-timer.C:
+							default:
+							}
+						}
 						return
-					case <-time.After(a.pollingInterval):
+					case <-timer.C:
+					case <-a.trigger.Triggered():
```

Comment:
Since all of the "sub-agents" share a single trigger, can we get into a situation where a notification is consumed by the wrong one? e.g. `Kick` is called for a variation request, but `<-a.trigger.Triggered()` is read by the goroutine for `runEdits` or `runGenerations` instead of `runVariations`.

This is probably not a big deal, since the timers ensure the request will be processed again shortly and all of the agents implemented with this "sub-agent" pattern support ancillary APIs, but I think it's still impactful enough to call out anyway.